src/System/Random/Internal.hs

```diff
@@ -103,20 +100,23 @@ class RandomGen g where
   -- @since 1.2
   genWord8 :: g -> (Word8, g)
   genWord8 = first fromIntegral . genWord32
+  {-# INLINE genWord8 #-}
```

Comment:
According to benchmarks I ran previously, some INLINEs decrease performance, so in previous rounds of optimisation I was very careful to add INLINE only where I could measure an improvement - but of course the benchmarks don't exercise all code paths, so it's difficult to draw absolute conclusions. Still, it would be interesting to see a before / after comparison.

Reply:
I verified our benchmarks and some inline pragmas indeed resulted in performance regressions, so thank you for catching it. I dropped the commit and verified benchmarks again, performance is back to normal ;)